### PR TITLE
update everything. Trim unused dependencies. Shift to extending React…

### PR DESCRIPTION
**Pull request by kentmclean**
….Component rather than using React.CreateClass. Etc.

### CODE REVIEW

**Comment by kentmclean** [comment]
Upon successful code review, a new npm version will need to be published

File: package.json [hunk 1]
```diff
@@ -20,29 +20,17 @@
...
   "devDependencies": {
-    "active-x-obfuscator": "0.0.2",
-    "autoprefixer-loader": "^1.2.0",
-    "babel": "^5.6.14",
-    "babel-core": "^4.7.13",
-    "babel-loader": "^4.2.0",
-    "css-loader": "^0.9.1",
-    "jasmine-core": "^2.1.3",
-    "jsx-loader": "^0.13.2",
-    "karma": "^0.12.31",
```
Comment (kentmclean) [inline package.json:31]: no tests in this repo, so all of these dependencies aren't being used

File: package.json [hunk 2]
```diff
@@ -20,29 +20,17 @@
...
-    "webpack": "^1.4.15"
+    "babel-cli": "^6.26.0",
+    "babel-loader": "^7.1.2",
+    "babel-plugin-transform-react-jsx": "^6.24.1",
+    "babel-preset-env": "^1.6.0",
+    "babel-preset-react": "^6.24.1",
+    "babel-preset-stage-2": "^6.24.1",
+    "css-loader": "^0.28.7",
+    "postcss-loader": "^2.0.6",
+    "react": "^16.0.0",
```
Comment (kentmclean) [inline package.json]: oh my!

File: webpack.config.js [hunk 3]
```diff
@@ -1,32 +1,41 @@
...
 var webpack = require('webpack');
-var ngminPlugin = require('ngmin-webpack-plugin');
 
 var version = require('./package.json').version;
 var name = require('./package.json').name;
 
 module.exports = {
 	entry: "./src/cpr-autoscale-input.js",
 	output: {
-		path: './build',
```
Comment (kentmclean) [inline webpack.config.js:11]: updated to bring it up to webpack 3.x.x compliance

File: src/cpr-autoscale-input.js [hunk 4]
```diff
@@ -4,18 +4,18 @@ var html = function(size, family) {
 	return `<span style="font-family:${family};font-size:${size}; font-weight:400;"></span>`;
 };
 
-var CanopyAutoscaleInput = React.createClass({
```
Comment (kentmclean) [inline src/cpr-autoscale-input.js:7]: sayonara, createClass!

File: package.json (hunk 2 above)
Comment (TheMcMurder) [reply package.json]: Hmmm... are we confident that we won't deal with any issues having this one use a different version of react even though we using webpack externals? I haven't played at all with React 16.

Comment (kentmclean) [reply package.json]: Nope! Not sure yet. I'll drop it back to the 15.x version

**Comment by TheMcMurder** [comment]
Looks good to me. 🐙

**Pull request merged**